src/plugins/visualizations/public/vis_types/vis_type_alias_registry.ts

```diff
@@ -64,19 +67,38 @@ export interface VisTypeAlias {
   };
 }
 
-const registry: VisTypeAlias[] = [];
+let registry: VisTypeAlias[] = [];
+let discardOnRegister: string[] = [];
 
 interface VisTypeAliasRegistry {
   get: () => VisTypeAlias[];
   add: (newVisTypeAlias: VisTypeAlias) => void;
+  remove: (visTypeAliasName: string) => void;
 }
 
 export const visTypeAliasRegistry: VisTypeAliasRegistry = {
   get: () => [...registry],
   add: (newVisTypeAlias) => {
-    if (registry.find((visTypeAlias) => visTypeAlias.name === newVisTypeAlias.name)) {
-      throw new Error(`${newVisTypeAlias.name} already registered`);
+    // if it exists on discardOnRegister array then we don't allow it to be registered
+    if (discardOnRegister.find((aliasName) => aliasName === newVisTypeAlias.name)) {
```

Comment:
Nitpick: Since we're actually only interested in whether the element exists and not retrieving it, we could use `some` instead of `find` here.